library/ssl_srv.c

```diff
@@ -2571,14 +2570,22 @@ static int ssl_write_certificate_request( mbedtls_ssl_context *ssl )
         /*
          * Supported signature algorithms
          */
+        for( cur = ssl->conf->sig_hashes; *cur != MBEDTLS_MD_NONE; cur++ )
+        {
+            unsigned char hash = mbedtls_ssl_hash_from_md_alg( *cur );
+
+            if( MBEDTLS_SSL_HASH_NONE == hash || mbedtls_ssl_set_calc_verify_md( ssl, hash ) )
+                continue;
+
 #if defined(MBEDTLS_RSA_C)
-        p[2 + sa_len++] = ssl->handshake->verify_sig_alg;
-        p[2 + sa_len++] = MBEDTLS_SSL_SIG_RSA;
+            p[2 + sa_len++] = hash;
```

Comment:
No I don't think so:
- If we would check it here, then we would have to check it at almost every line in `ssl_tls.c`, `ssl_cli.c`, `ssl_srv.c`
- the `ssl->conf->sig_hashes` structure is controlled solely by the application, remote actors have no effect on it
- as you mentioned it too: the buffer is very large and the chances of overflow are quite slim